packages/components/src/settings-form/types.ts

```diff
@@ -0,0 +1,32 @@
+export interface StringToString {
+	[ key: string ]: string;
+}
+
+export interface Field {
+	id: string;
+	type: 'text' | 'password' | 'checkbox' | 'select';
+	title: string;
+	label: string;
+	description?: string;
+	default?: string;
+	class?: string;
+	css?: string;
+	options?: StringToString;
+	tip?: string;
+	value?: string;
+	placeholder?: string;
+}
+
+export interface GetInputPropsReturn {
+	value: string;
```

Comment:
A lot of these fields look like props available intrinsically on html elements. You should take a look at the utility types React has for these.

One thats really useful is `React.HtmlAttributes<>`. If you pass it one of the element types you can specify what element specific properties you want to target, which would be useful in differentiating the different kind of form elements present.

here's an example where I refactored our Badge component, which is wrapped with a span:

```
type Props = {
	count: number;
} & React.HTMLAttributes< HTMLSpanElement >;

export const Badge: React.FC< Props > = ( {
	count,
	className = '',
	...props
}: Props ): JSX.Element => {
	return (
		<span className={ `woocommerce-badge ${ className }` } { ...props }>
			{ count }
		</span>
	);
};
```

Reply:
Thanks, I was a little unclear on these since they're generated by the `<Form />` component, but it _does_ look like it would be safe to utilize a union with a built-in type for many of those. 👍🏻